addons/hearing/XEH_postInit.sqf

```diff
@@ -16,16 +16,10 @@ if (isServer) then {
     // Only install event handler if combat deafness is enabled
     if (!GVAR(enableCombatDeafness)) exitWith {};
 
-    addMissionEventHandler ["ProjectileCreated", {
-        params ["_projectile"];
-
-        if (!local _projectile) exitWith {};
-
-        // Rockets only explode on local clients
-        _projectile addEventHandler ["Explode", {
-            [QGVAR(explosion), _this] call CBA_fnc_globalEvent;
-        }];
-    }];
+    [{ // Convert ace_common's local explosion to a hearing global explosion event 
+        TRACE_1("Explode",_this);
+        [QGVAR(explosion), _this select [0,2]] call CBA_fnc_globalEvent; // trim unused 3rd arg for network savings
```

Comment:
Could also add a `isNull` check, but imo not necessary, since `objNull` usually doesn't explode.

```suggestion
        params ["_projectile", "_pos"];
        TRACE_1("Explode",_this);
        
        // If projectile is local only, don't raise event globally
        if (_projectile call BIS_fnc_netId == "0:0") then { // TODO: Use netId instead of BIS_fnc_netId in 2.18
            [QGVAR(explosion), [_projectile, _pos]] call CBA_fnc_localEvent;
        } else {
            [QGVAR(explosion), [_projectile, _pos]] call CBA_fnc_globalEvent;
        };
```

Reply:
I'd really prefer to handle this in the common func
and present a clean local only event for other addons to use
but I can't see anyway of doing that until 2.18

Reply:
`_projectile call BIS_fnc_netId == "0:0"` should work for that too then.

Reply:
all machines see netId of "0:0"
I think only isReal will be able to tell who is the actual "owner"

Reply:
> all machines see netId of "0:0" I think only isReal will be able to tell who is the actual "owner"

With your comment, I *assume* that `isReal` returns `true` on a single machine (I have no idea how it works and haven't tested). I previously assumed it would return `false` if it was a projectile local to all clients.

If my assumption is true, raising the event on the machine with the `isReal` flag only would break hearing in MP, no? I'm guessing said projectile is local to that machine only, so we can't pass it globally.